tfe/resource_tfe_workspace_variable_set.go

```diff
@@ -0,0 +1,183 @@
+
+	log.Printf("[DEBUG] Read configuration of workspace variable set: %s", d.Id())
+	vS, err := tfeClient.VariableSets.Read(ctx, vSID, &tfe.VariableSetReadOptions{
+		Include: &[]tfe.VariableSetIncludeOpt{tfe.VariableSetWorkspaces},
+	})
+	if err != nil {
+		if errors.Is(err, tfe.ErrResourceNotFound) {
+			log.Printf("[DEBUG] Variable set %s no longer exists", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading configuration of variable set %s: %w", d.Id(), err)
+	}
+
+	// Verify workspace listed in variable set
+	check := false
+	for _, workspace := range vS.Workspaces {
+		if workspace.ID == wID {
+			check = true
+			d.Set("workspace_id", wID)
+		}
+	}
+	if !check {
+		log.Printf("[DEBUG] Workspace %s not attached to variable set %s. Removing from state.", wID, vSID)
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("variable_set_id", vSID)
+	return nil
+}
+
+func resourceTFEWorkspaceVariableSetDelete(d *schema.ResourceData, meta interface{}) error {
+	tfeClient := meta.(*tfe.Client)
+
+	wID := d.Get("workspace_id").(string)
+	vSID := d.Get("variable_set_id").(string)
+
+	log.Printf("[DEBUG] Delete workspace (%s) from variable set (%s)", wID, vSID)
+	removeOptions := tfe.VariableSetRemoveFromWorkspacesOptions{}
+	removeOptions.Workspaces = append(removeOptions.Workspaces, &tfe.Workspace{ID: wID})
+
+	err := tfeClient.VariableSets.RemoveFromWorkspaces(ctx, vSID, &removeOptions)
+	if err != nil {
+		return fmt.Errorf(
+			"Error removing workspace %s from variable set %s: %w", wID, vSID, err)
+	}
+
+	return nil
+}
+
+func encodeVariableSetWorkspaceAttachment(wID, vSID string) string {
+	return fmt.Sprintf("%s_%s", wID, vSID)
+}
+
+func resourceTFEWorkspaceVariableSetImporter(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	// The format of the import ID is <ORGANIZATION/WORKSPACE NAME/VARSET NAME> but be aware
+	// that variable set names can contain forward slash characters but organization/workspace
+	// names cannot. Therefore, we split the import ID into at most 3 substrings.
+	organization, wsName, vSName, err := destructureImportID(strings.SplitN(d.Id(), "/", 3))
+	if err != nil {
+		return nil, err
+	}
+
+	tfeClient := meta.(*tfe.Client)
+
+	// Ensure a workspace of this name exists before fetching all the variable sets in the org
+	_, err = tfeClient.Workspaces.Read(ctx, organization, wsName)
+	if err != nil {
+		return nil, fmt.Errorf("error reading configuration of workspace %s in organization %s: %w", wsName, organization, err)
+	}
+
+	options := &tfe.VariableSetListOptions{
+		Include: string(tfe.VariableSetWorkspaces),
+	}
+	for {
+		list, err := tfeClient.VariableSets.List(ctx, organization, options)
+		if err != nil {
+			return nil, fmt.Errorf("Error retrieving variable sets: %w", err)
+		}
+		for _, vs := range list.Items {
+			if vs.Name != vSName {
+				continue
+			}
+
+			for _, ws := range vs.Workspaces {
+				if ws.Name != wsName {
+					break
```

Comment:
```suggestion
					continue
```

Import didn't work as described, so I went digging. This looks to be the culprit!

Reply:
Yikes! In my smoke test it must have been the first result 😬